### PR TITLE
uvicorn 0.23.2 (new formula)

### PR DESCRIPTION
**Pull request by singingwolfboy**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/u/uvicorn.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    curl_cmd = "curl --silent 127.0.0.1:#{port}/"
+    ohai curl_cmd
+    output = ""
+    exitstatus = 7
+    loop do
+      sleep 1
+      output = `#{curl_cmd}`
+      exitstatus = $CHILD_STATUS.exitstatus
+      break if exitstatus != 7  # CURLE_COULDNT_CONNECT
+    end
```
Comment (SMillerDev) [inline Formula/u/uvicorn.rb:100]: Why loop and not just wait?

File: Formula/u/uvicorn.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    EOS
+
+    port = free_port
+    pid = fork do
+      exec bin/"uvicorn", "--port=#{port}", "example:app"
+    end
+
+    # keep trying to connect until the server is up
+    curl_cmd = "curl --silent 127.0.0.1:#{port}/"
+    ohai curl_cmd
```
Comment (SMillerDev) [inline Formula/u/uvicorn.rb:92]: We shouldn't do `ohai` in formulae

File: Formula/u/uvicorn.rb (hunk 1 above)
Comment (singingwolfboy) [reply Formula/u/uvicorn.rb]: I copied this from the `moto` formula -- but I see other formulae just do `sleep 1` and that works. I changed this formula to do that, too.

File: Formula/u/uvicorn.rb (hunk 2 above)
Comment (singingwolfboy) [reply Formula/u/uvicorn.rb]: I also copied this from the `moto` formula -- I guess we should remove it from that formula.

File: Formula/u/uvicorn.rb (hunk 1 above)
Comment (singingwolfboy) [reply Formula/u/uvicorn.rb:99]: Update; just doing `sleep 1` does _not_ work reliably. It introduces a race condition; if the server does not get running by the time the fixed sleep is done, the test will fail. Looping prevents this race condition.

Comment (SMillerDev) [reply Formula/u/uvicorn.rb:99]: Yes, that's a bit of a gamble but it is the default in many formula. If 1 doesn't always work just set it to 10.

**Review by SMillerDev — APPROVED** [review]
Thanks @singingwolfboy  ! Without your contributions it'd be impossible to keep homebrew going. You can feel good knowing that you've made the world a lot better for homebrew users around the world! 👍 🎉

**Pull request merged**